modules/client/src/main/java/org/apache/ignite/client/IgniteClient.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client;
+
+import org.apache.ignite.app.Ignite;
+import org.apache.ignite.internal.client.TcpIgniteClient;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Ignite client entry point.
+ */
+public class IgniteClient {
+    /**
+     * Gets a new client builder.
+     *
+     * @return New client builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /** Client builder. */
+    public static class Builder {
+        /** Addresses. */
+        private String[] addresses;
+
+        /**
+         * Builds the client.
+         *
+         * @return Ignite client.
+         */
+        public Ignite build() {
```

Comment:
Nu sure if I understand the point. `Ignite` is an entry point of our API, no matter thin client or server (inside Compute and so on). The idea is to have the same API in all cases - same set of interfaces.

Reply:
On the second thought, it looks OK. I meant that `Ignite` in 2.x means interface to local node, while here it is Client's interface, but well, I think it's OK since we changing a concept.